### PR TITLE
docs: clarify usage in system-assigned mode

### LINKED ISSUE #388: System-assigned Managed Identity
[This page](https://azure.github.io/secrets-store-csi-driver-provider-azure/configurations/identity-access-modes/system-assigned-msi-mode/) says -

> Configure System-assigned Managed Identity to access Keyvault 
AKS uses system-assigned managed identity as cluster managed identity. This managed identity shouldn’t be used to access Keyvault. **You should consider using a User-assigned managed identity instead.**

** emphasis mine.

That's great but it does not address the WHY? Why is one better than the other and what are the risks if one uses managed identity instead?

### LINKED ISSUE #389: Standard Walkthrough
This page -
https://azure.github.io/secrets-store-csi-driver-provider-azure/demos/standard-walkthrough/#5-deploy-secretproviderclass

references `userAssignedIdentityID: ""` which isn't clear what it is. It's not the resource id (which would be my first choice just by reading the key name) but the Client ID.

You should add a comment above it to clearly state that's the CLIENT ID of the user identity, so just the GUID value.

### PR DESCRIPTION
**Pull request by aramase**
Signed-off-by: Anish Ramasekar <anish.ramasekar@gmail.com>

**Reason for Change**:

- Update system-assigned managed identity mode doc
  - The system-assigned managed identity created by AKS is not part of the node VM/VMSS. This means users can create their own system-assigned managed identity and use to access keyvault. Also, pinged @miwithro to update the AKS docs to say the same when creating a cluster with managed identity. Even though the cluster identity profile shows `type: SystemAssignedIdentity`, there is no system-assigned managed identity in the VM/VMSS.

**Requirements**

- [x] squashed commits
- [x] included documentation
- [ ] added unit tests and e2e tests (if applicable).

**Issue Fixed**:

fixes #388 
fixes #389 

**Does this change contain code from or inspired by another project?**

- [ ] Yes
- [ ] No

**If "Yes," did you notify that project's maintainers and provide attribution?**

**Special Notes for Reviewers**:

### CODE REVIEW

File: website/content/en/configurations/identity-access-modes/system-assigned-msi-mode.md [hunk 1]
```diff
@@ -68,9 +68,7 @@ spec:
 
 ## Configure System-assigned Managed Identity to access Keyvault
 
-AKS uses system-assigned managed identity as [cluster managed identity](https://docs.microsoft.com/en-us/azure/aks/use-managed-identity). This managed identity shouldn't be used to access Keyvault. You should consider using a [User-assigned managed identity](./user-assigned-msi-mode) instead.
-
-Before this step, you need to turn on system-assigned managed identity on your cluster VM/VMSS.
+Before this step, you need to [enable system-assigned managed identity](https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/qs-configure-cli-windows-vm#enable-system-assigned-managed-identity-on-an-existing-azure-vm) in your cluster VM/VMSS.
```
Comment (ritazh) [inline website/content/en/configurations/identity-access-modes/system-assigned-msi-mode.md:71]: > This managed identity shouldn't be used to access Keyvault. You should consider using a [User-assigned managed identity](./user-assigned-msi-mode) instead.
Is this guidance no longer valid?

Comment (aramase) [reply website/content/en/configurations/identity-access-modes/system-assigned-msi-mode.md:71]: I confirmed this was never the case. There is no system-assigned managed identity created in the cluster. I've added some context in the PR description.

Comment (ritazh) [reply website/content/en/configurations/identity-access-modes/system-assigned-msi-mode.md:71]: Yea I saw the PR description. I guess the key for this PR is to guide users to create system msi on the node and remove any reference about the cluster managed identity right?

Comment (aramase) [reply website/content/en/configurations/identity-access-modes/system-assigned-msi-mode.md:71]: yeah, that's right. The cluster managed identity is a user-assigned managed identity, so we're stubbing that out of the doc. I've added a doc ref on how to enable system-assigned managed identity. The rest of the doc shows how to verify the system-assigned managed identity exists and role assignments.

**Review by ritazh — APPROVED** [review]

**Pull request merged**